pkg/ingresscache/safetrie.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2025 Iguazio Systems Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License") with
+an addition restriction as set forth herein. You may not use this
+file except in compliance with the License. You may obtain a copy of
+the License at http://www.apache.org/licenses/LICENSE-2.0.
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing
+permissions and limitations under the License.
+
+In addition, you may not use the software for any purposes that are
+illegal under applicable law, and the grant of the foregoing license
+under the Apache 2.0 license is conditioned upon your compliance with
+such restriction.
+*/
+
+package ingresscache
+
+import (
+	"slices"
+	"sync"
+
+	"github.com/dghubble/trie"
+	"github.com/nuclio/errors"
+)
+
+type SafeTrie struct {
+	pathTrie *trie.PathTrie
+	rwMutex  sync.RWMutex
+}
+
+// NewSafeTrie creates a new SafeTrie instance
+func NewSafeTrie() *SafeTrie {
+	return &SafeTrie{
+		pathTrie: trie.NewPathTrie(),
+		rwMutex:  sync.RWMutex{},
+	}
+}
+
+// SetFunctionName sets a function for a given path. If the path does not exist, it creates it
+func (st *SafeTrie) SetFunctionName(path string, function string) error {
+	if path == "" {
+		return errors.New("path is empty")
+	}
+
+	if function == "" {
+		return errors.New("function is empty")
+	}
+
+	st.rwMutex.Lock()
+	defer st.rwMutex.Unlock()
+
+	// get the exact path value in order to avoid creating a new path if it already exists
+	pathValue := st.pathTrie.Get(path)
+	if pathValue == nil {
+		st.pathTrie.Put(path, []string{function})
+		return nil
+	}
+
+	pathFunctionNames, ok := pathValue.([]string)
+	if !ok {
+		return errors.Errorf("value is not a []string, got: %T", pathValue)
```

Comment:
```suggestion
		return errors.Errorf("path value should be []string, got %T", pathValue)
```
to align with the error in Delete function

Reply:
Fixed here - [CR comment - UTs (renamings and reverting SetupTest for logger) and n…](https://github.com/v3io/scaler/pull/70/commits/31993658671fbed076572d370cffd069493bd94f)